File: C08_WiimoteCar/WiimoteCarVB/Form1.vb
Imports System
Imports System.Windows.Forms
Imports Phidgets
Imports WiimoteLib

Namespace WiimoteCar
	Partial Public Class Form1
		Inherits Form
		' instance of the connected Wiimote
		Private _wiimote As Wiimote

		' instance of the connected Phidget Interface Kit
		Private _interfaceKit As InterfaceKit

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' create the InterfaceKit object and open access to the board
			_interfaceKit = New InterfaceKit()
			_interfaceKit.open()
			_interfaceKit.waitForAttachment(5000)

			' create the Wiimote object
			_wiimote = New Wiimote()

			' setup an event handler to be notified when the Wiimote sends a packet of data
			AddHandler _wiimote.WiimoteChanged, AddressOf Wiimote_WiimoteChanged

			' connect the Wiimote
			_wiimote.Connect()

			' set the report type to Buttons and Accelerometer data only
			_wiimote.SetReportType(InputReport.ButtonsAccel, True)
		End Sub

		Private Sub Wiimote_WiimoteChanged(ByVal sender As Object, ByVal e As WiimoteChangedEventArgs)
			' get the current Wiimote state
			Dim ws As WiimoteState = e.WiimoteState

			' if button 1 is pressed, toggle the Forward output
			_interfaceKit.outputs(0) = ws.ButtonState.One

			' if button 2 is pressed, toggle the Backward output
			_interfaceKit.outputs(1) = ws.ButtonState.Two

			' if the Wiimote is tilted far enough to the left, toggle the Left output
			_interfaceKit.outputs(2) = (ws.AccelState.Values.Y < -0.07f)

			' if the Wiimote is tilted far enough to the right, toggle the Reft output
			_interfaceKit.outputs(3) = (ws.AccelState.Values.Y > 0.07f)
		End Sub
	End Class
End Namespace

File: C8_WiimoteCar/WiimoteCarCS/Form1.cs
using System;
using System.Windows.Forms;
using Phidgets;
using WiimoteLib;

namespace WiimoteCar
{
	public partial class Form1 : Form
	{
		// instance of the connected Wiimote
		private Wiimote _wiimote;

		// instance of the connected Phidget Interface Kit
		private InterfaceKit _interfaceKit;

		public Form1()
		{
			InitializeComponent();
		}

		private void Form1_Load(object sender, EventArgs e)
		{
			// create the InterfaceKit object and open access to the board
			_interfaceKit = new InterfaceKit();
			_interfaceKit.open();
			_interfaceKit.waitForAttachment(1000);

			// create the Wiimote object
			_wiimote = new Wiimote();

			// setup an event handler to be notified when the Wiimote sends a packet of data
			_wiimote.WiimoteChanged += Wiimote_WiimoteChanged;

			// connect the Wiimote
			_wiimote.Connect();

			// set the report type to Buttons and Accelerometer data only
			_wiimote.SetReportType(InputReport.ButtonsAccel, true);
		}

		void Wiimote_WiimoteChanged(object sender, WiimoteChangedEventArgs e)
		{
			// get the current Wiimote state
			WiimoteState ws = e.WiimoteState;

			// if button 1 is pressed, toggle the Forward output
			_interfaceKit.outputs[0] = ws.ButtonState.One;

			// if button 2 is pressed, toggle the Backward output
			_interfaceKit.outputs[1] = ws.ButtonState.Two;

			// if the Wiimote is tilted far enough to the left, toggle the Left output
			_interfaceKit.outputs[2] = (ws.AccelState.Values.Y < -0.07f);

			// if the Wiimote is tilted far enough to the right, toggle the Reft output
			_interfaceKit.outputs[3] = (ws.AccelState.Values.Y >  0.07f);
		}
	}
}
